app/controllers/learning_materials_controller.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class LearningMaterialsController < ApplicationController
+  skip_before_action :authenticate_user!, only: %i[index]
+
+  def index
+    @learningmaterials = LearningMaterial.all
+    @learningmaterials_with_thumbnails = @learningmaterials.select do |learningmaterial|
+      learningmaterial.thumbnail.attached?
+    end
+    @random_learningmaterials = @learningmaterials_with_thumbnails.sample(2)
+    @learningmaterials_with_or_without_thumbnails = @learningmaterials - @random_learningmaterials
```

Comment:
```suggestion
    @learning_materials_without_random = @learning_materials - @random_learning_materials
```
maybe this makes more sense? 🤷‍♂️ naming things is hard 😅

Reply:
Thank you for this suggestion and yes naming things is really hard because there is never a "one  right/ true way" to name things. The reason why it is named as "with or without thumbnails" is because it is supposed to hold all other Learning materials regardless of whether that specific learning material has an attached thumbnail or not. If this is renamed to your suggestion I am afraid it will lead to slight confusion later.

What do you think @denissellu ? :thinking: